src/Messaging/Messaging.ts

```diff
@@ -4,15 +4,17 @@ import { Connection, connect, Channel } from 'amqplib'
 
 import { ClaimIPFSHashPair, isClaimIPFSHashPair } from 'Interfaces'
 
-import { Exchange } from './Messages'
+import { ExchangeConfiguration } from './ExchangeConfiguration'
 
 export class Messaging {
   private readonly connectionUrl: string
   private connection: Connection
   private channel: Channel
+  private readonly exchanges: ExchangeConfiguration
 
-  constructor(connectionUrl?: string) {
-    this.connectionUrl = connectionUrl || 'amqp://localhost'
+  constructor(connectionUrl = 'amqp://localhost', exchanges: ExchangeConfiguration) {
```

Comment:
Let's not default this here because that is the responsibility of the Configuration defaults. If this is not set, we want to app to fail.